pkg/agent/stats/collector.go

```diff
@@ -168,7 +175,43 @@ func (m *Collector) report(curStatsCollection *statsCollection) error {
 	npStats := calculateDiff(curStatsCollection.networkPolicyStats, m.lastStatsCollection.networkPolicyStats)
 	acnpStats := calculateRuleDiff(curStatsCollection.antreaClusterNetworkPolicyStats, m.lastStatsCollection.antreaClusterNetworkPolicyStats)
 	anpStats := calculateRuleDiff(curStatsCollection.antreaNetworkPolicyStats, m.lastStatsCollection.antreaNetworkPolicyStats)
-	if len(npStats) == 0 && len(acnpStats) == 0 && len(anpStats) == 0 {
+
+	multicastGroups := []cpv1beta.MulticastGroupInfo{}
+	if m.multicastEnabled() {
+		multicastGroupMap := m.multicastQuerier.GetGroupPods()
+		multicastGroups = make([]cpv1beta.MulticastGroupInfo, 0, len(multicastGroupMap))
+		for group, pods := range multicastGroupMap {
+			multicastGroups = append(multicastGroups, cpv1beta.MulticastGroupInfo{Group: group, Pods: pods})
+		}
+
+		// Collect statistics of IGMP report messages hit by ANP or ACNP, and merge them to anpStats and acnpStats.
+		multicastStatsMap := m.multicastQuerier.GetIGMPReportNPStats()
+		mergeReportStats := func(igmpReportStatsMap map[types.UID]map[string]*agenttypes.RuleMetric, originalStatsList []cpv1beta.NetworkPolicyStats) []cpv1beta.NetworkPolicyStats {
+			uidIndexMap := make(map[types.UID]int)
+			for i, stats := range originalStatsList {
+				uidIndexMap[stats.NetworkPolicy.UID] = i
+			}
+			for uid, npStats := range igmpReportStatsMap {
+				ruleStatsList := make([]statsv1alpha1.RuleTrafficStats, 0, len(npStats))
+				for ruleName, ruleStats := range npStats {
+					ruleStatsList = append(ruleStatsList, statsv1alpha1.RuleTrafficStats{Name: ruleName, TrafficStats: statsv1alpha1.TrafficStats{Packets: int64(ruleStats.Packets), Bytes: int64(ruleStats.Bytes)}})
+				}
+				index, exist := uidIndexMap[uid]
+				if !exist {
+					originalStatsList = append(originalStatsList, cpv1beta.NetworkPolicyStats{NetworkPolicy: cpv1beta.NetworkPolicyReference{UID: uid}, RuleTrafficStats: ruleStatsList})
+				} else {
+					originalStatsList[index].RuleTrafficStats = append(originalStatsList[index].RuleTrafficStats, ruleStatsList...)
+				}
+			}
+			return originalStatsList
+		}
+
+		acnpStats = mergeReportStats(multicastStatsMap[cpv1beta.AntreaClusterNetworkPolicy], acnpStats)
+		anpStats = mergeReportStats(multicastStatsMap[cpv1beta.AntreaNetworkPolicy], anpStats)
+
+	}
+
+	if len(npStats) == 0 && len(acnpStats) == 0 && len(anpStats) == 0 && len(multicastGroups) == 0 {
```

Comment:
Updated. Thank you for pointing that out. I have changed the implementation here to only report the group members if group members have changed since the last report